Console.rb

```diff
@@ -0,0 +1,61 @@
+# Utility for writing to terminal, mostly colors and cursor movements.
+
+module Console
+	def self.color(color, text)
+		return text if color.nil?
+		self.public_send(color, text)
+	end
+
+  def self.red(text)
+    color_encode(text, 31)
+  end
+
+  def self.green(text)
+    color_encode(text, 32)
+  end
+
+  def self.blue(text)
+		color_encode(text, 34)
+  end
+
+  def self.cyan(text)
+    color_encode(text, 36)
+  end
+
+  def self.magenta(text)
+    color_encode(text, 35)
+  end
+
+  def self.light_green(text)
+    color_encode(text, 92)
+  end
+
+  def self.color_encode(text, code)
+    "\e[#{code}m#{text}\e[0m"
+  end
+
+  def self.emph(text) # bold underline
+		"\e[1m\e[4m#{text}"
+  end
+
+  def self.left(num)
+		"\033[#{num}D"
```

Comment:
Use of `tput` would theoretically increase compatibilty. I don't have much Ruby experience, but it seems like use of backticks to execute a system command and get its output would be pretty convenient, and just as terse as what you're doing now.

The main downside is that calling another process for every keystroke will be "expensive" (but probably still imperceptible to the user). If that is an issue, `tput` could be called once at initialization and the output saved for later use in these methods.